### ISSUE TITLE
2023 Roadmaps - I'm back!

### ISSUE THREAD

**Issue by saltyshiomix** [saltyshiomix/nextron#398]
## I'm back!

Suffering from bipolar disorder, I was unable to maintain nextron for over 7 months. Medication has allowed me to be somewhat more active, and I will be gradually improving this project.

I would like to thank everyone who has supported nextron so far:

- All stargazers
- All used by
- All contributers
- All collaborators
- All sponsors

Thank you very much for your continued support.

### Why nextron?

- https://github.com/saltyshiomix/nextron/issues/92
- https://github.com/saltyshiomix/nextron/issues/233

### Core features

- https://github.com/saltyshiomix/nextron/issues/247
- https://github.com/saltyshiomix/nextron/issues/300
- https://github.com/saltyshiomix/nextron/pull/362

### Bugs

- https://github.com/saltyshiomix/nextron/issues/121
- https://github.com/saltyshiomix/nextron/issues/168
- https://github.com/saltyshiomix/nextron/issues/170
- https://github.com/saltyshiomix/nextron/issues/279
- https://github.com/saltyshiomix/nextron/issues/241
- https://github.com/saltyshiomix/nextron/issues/291
- https://github.com/saltyshiomix/nextron/issues/299
- https://github.com/saltyshiomix/nextron/issues/307
- https://github.com/saltyshiomix/nextron/issues/309
- https://github.com/saltyshiomix/nextron/issues/346
- https://github.com/saltyshiomix/nextron/issues/384

### Next v13 support & Nextron v9 release

- https://github.com/saltyshiomix/nextron/issues/296
- https://github.com/saltyshiomix/nextron/issues/299
- https://github.com/saltyshiomix/nextron/issues/339

### Working with databases

- https://github.com/saltyshiomix/nextron/issues/253
- https://github.com/saltyshiomix/nextron/issues/386

### Improve [examples](https://github.com/saltyshiomix/nextron/tree/main/examples)

- https://github.com/saltyshiomix/nextron/issues/94
- https://github.com/saltyshiomix/nextron/pull/356
- https://github.com/saltyshiomix/nextron/issues/377
- https://github.com/saltyshiomix/nextron/issues/378
- https://github.com/saltyshiomix/nextron/issues/391
- https://github.com/saltyshiomix/nextron/issues/395
- https://github.com/saltyshiomix/nextron/issues/381
- https://github.com/saltyshiomix/nextron/issues/376
- https://github.com/saltyshiomix/nextron/issues/374

### Questions and "check back later"

These questions may be the subject of future development.

- https://github.com/saltyshiomix/nextron/issues/65
- https://github.com/saltyshiomix/nextron/issues/82
- https://github.com/saltyshiomix/nextron/issues/85
- https://github.com/saltyshiomix/nextron/issues/86
- https://github.com/saltyshiomix/nextron/issues/133
- https://github.com/saltyshiomix/nextron/issues/140
- https://github.com/saltyshiomix/nextron/issues/142
- https://github.com/saltyshiomix/nextron/issues/152
- https://github.com/saltyshiomix/nextron/issues/155
- https://github.com/saltyshiomix/nextron/issues/169
- https://github.com/saltyshiomix/nextron/issues/171
- https://github.com/saltyshiomix/nextron/issues/200
- https://github.com/saltyshiomix/nextron/issues/231
- https://github.com/saltyshiomix/nextron/issues/234
- https://github.com/saltyshiomix/nextron/issues/257
- https://github.com/saltyshiomix/nextron/issues/266
- https://github.com/saltyshiomix/nextron/issues/267
- https://github.com/saltyshiomix/nextron/issues/269
- https://github.com/saltyshiomix/nextron/issues/280
- https://github.com/saltyshiomix/nextron/issues/282
- https://github.com/saltyshiomix/nextron/issues/292
- https://github.com/saltyshiomix/nextron/issues/302
- https://github.com/saltyshiomix/nextron/issues/308
- https://github.com/saltyshiomix/nextron/issues/317
- https://github.com/saltyshiomix/nextron/issues/319
- https://github.com/saltyshiomix/nextron/issues/321
- https://github.com/saltyshiomix/nextron/issues/323
- https://github.com/saltyshiomix/nextron/issues/333
- https://github.com/saltyshiomix/nextron/issues/340
- https://github.com/saltyshiomix/nextron/issues/341
- https://github.com/saltyshiomix/nextron/issues/343
- https://github.com/saltyshiomix/nextron/issues/352
- https://github.com/saltyshiomix/nextron/issues/353
- https://github.com/saltyshiomix/nextron/issues/355
- https://github.com/saltyshiomix/nextron/issues/363
- https://github.com/saltyshiomix/nextron/issues/365
- https://github.com/saltyshiomix/nextron/issues/368
- https://github.com/saltyshiomix/nextron/issues/369
- https://github.com/saltyshiomix/nextron/issues/396

## Good Issues

- TypeScript paths
  - https://github.com/saltyshiomix/nextron/issues/235
  - https://github.com/saltyshiomix/nextron/issues/270
- Environment variables
  - https://github.com/saltyshiomix/nextron/issues/128
  - https://github.com/saltyshiomix/nextron/issues/188
  - https://github.com/saltyshiomix/nextron/issues/157
- SSR support? No.
  - https://github.com/saltyshiomix/nextron/issues/263

### Help Wanted

- https://github.com/saltyshiomix/nextron/issues/132
- https://github.com/saltyshiomix/nextron/issues/141
- https://github.com/saltyshiomix/nextron/issues/336

## FYI

- https://github.com/saltyshiomix/nextron/discussions/244

### Useful references

- https://github.com/vercel/next.js/tree/canary/examples

**Comment by makeusabrew** [comment]
Hi @saltyshiomix! I don't know you personally, but I just wanted to say welcome back. Nextron is a great project and you've done some great work.

The list of issues you've posted above looks pretty daunting, so if there's anything the community can do to help with them (even if it's help decide if they can just be closed?), shout. I don't have bags of free time, but I'll try and do my bit and I'm sure others will too.

**Comment by saltyshiomix** [comment]
@makeusabrew 

Thank you very much, and you are totally right!

I'll improve nextron step by step, too 🙇

**Comment by bm777** [comment]
I'm looking forward to it because I'm building my startup product with Nextron because I have struggled a bit with some features to implement in Nextron with external links.

I will distribute the beta version to people to test it before shipping the first release.

![Screenshot 2023-09-24 at 09 15 46](https://github.com/saltyshiomix/nextron/assets/29865600/be685c76-5084-4d32-8db6-e5159e3f3b51)

Thanks, @saltyshiomix  for creating Nextron ✌️

**Comment by saltyshiomix** [comment]
@bm777 

Thank you for using nextron!
I appreciate it very much.

Please feel free to open issues if you have any troubles :)

**Comment by bm777** [comment]
@saltyshiomix 

I will definitely :-)

I think, I will open an issue today, if we solve that issue, it will be a great achievement.

**Comment by saltyshiomix** [comment]
@bm777 

I just created the disscussion:

- https://github.com/saltyshiomix/nextron/discussions/406

Feel free to share your apps :)

**Comment by bm777** [comment]
@saltyshiomix 

Yes of course, when the beta is ready I will share it with you first to correct some bugs 😉

**Comment by riclava** [comment]
Electron 22+ is nice ;-)

**Comment by saltyshiomix** [comment]
@riclava 
Thanks!
To use electron v22+, please see this introduction:
https://github.com/saltyshiomix/nextron/issues/299#issuecomment-1731438336

**Comment by saltyshiomix** [comment]
Nextron v8.12.0 is out!

Now it supports preload script for secure IPC features :)

**Comment by alexis-piquet** [comment]
Nice to see you back in the game @saltyshiomix. Have you plan to bump next version to 14? 'cause it's the latest available version. Let me know, I would like help you on this subject ;)

**Comment by saltyshiomix** [comment]
@alexis-piquet 

Sorry for late reply.

Yes, I'm planning nextron v9 which supports both next v13 and v14 🙇

**Comment by pixelass** [comment]
Welcome back.. life is `****`, the community is great. 

BTW we use nextjs 13 with i18next (with no real issues). Happy to provide a basic example if it helps.

**Comment by saltyshiomix** [comment]
@pixelass 

I appreciate very much your sponsorship!

Thank you also for the pull request, I'll check it out soon :)

**Comment by saltyshiomix** [comment]
I apologize for the very slow work on support Next.js v14. I'm preparing it little by little, so please wait a little longer.

**Comment by saltyshiomix** [comment]
CC: @makeusabrew @bm777 @riclava @alexis-piquet @pixelass 

Just released nextron v5.15.0, which supports Next.js v13 🎉 

Next.js v14 support is coming soon 🙇

**Comment by bm777** [comment]
Oh waouh, Amazing new :) @saltyshiomix

**Comment by saltyshiomix** [comment]
I just released nextron v9.0.0, which supports next.js v13 and v14 🎉 

NOTE: Please update renderer/next.config.js as belows:

```js
module.exports = {
  output: 'export',
  // we want to change distDir to "app" so as nextron can build the app in production mode!
  distDir: process.env.NODE_ENV === 'production' ? '../app' : '.next',
  trailingSlash: true,
  images: {
    unoptimized: true,
  },
}
```

**Comment by saltyshiomix** [comment]
Nextron v10 is out, so I close this issue for now :)

https://npm.im/nextron
